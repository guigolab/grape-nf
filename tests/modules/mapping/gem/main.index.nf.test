nextflow_process {

    name "Test Process index"
    script "modules/mapping/gem/main.nf"
    process "index"

    tag "module"

    test("Should create index using GEM (compressed references)") {

        when {
            process {
                """
                input[0] = file("${baseDir}/data/genome.fa.gz")
                input[1] = file("${baseDir}/data/annotation.gtf.gz")
                """
            }
        }

        then {
            assert process.success
            with(process.out) {
                def genomeDir = get(0)[0]
                assert path(genomeDir).list().size() == 7
                assert snapshot(
                    path("${genomeDir}/genome_index.gem"),
                    path("${genomeDir}/transcript_index.junctions"),
                    path("${genomeDir}/transcript_index.junctions.fa"),
                    path("${genomeDir}/transcript_index.junctions.gem"),
                    path("${genomeDir}/transcript_index.junctions.keys"),
                ).match()
            }
        }

    }

    test("Should create index using GEM (uncompressed references)") {

        when {
            process {
                """
                input[0] = file("${baseDir}/data/genome.fa")
                input[1] = file("${baseDir}/data/annotation.gtf")
                """
            }
        }

        then {
            assert process.success
            with(process.out) {
                def genomeDir = get(0)[0]
                assert path(genomeDir).list().size() == 7
                assert snapshot(
                    path("${genomeDir}/genome_index.gem"),
                    path("${genomeDir}/transcript_index.junctions"),
                    path("${genomeDir}/transcript_index.junctions.fa"),
                    path("${genomeDir}/transcript_index.junctions.gem"),
                    path("${genomeDir}/transcript_index.junctions.keys"),
                ).match()
            }
        }

    }

}
