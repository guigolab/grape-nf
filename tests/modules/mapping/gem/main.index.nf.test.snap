{
    "Should create index using GEM (compressed references)": {
        "content": [
            "genome_index.gem:md5,ebd155f05251ace3a35ce3edd8816211",
            "transcript_index.junctions:md5,a0433e39f0c35e798609ff15332e12d4",
            "transcript_index.junctions.fa:md5,be0a28c75266e6969c11e308341810ae",
            "transcript_index.junctions.gem:md5,b53971e7b5358fd8cae8bb7f2a4a729d",
            "transcript_index.junctions.keys:md5,34985e55c280acd027d2dd32906ad59c"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.0"
        },
        "timestamp": "2024-05-02T09:01:47.328374722"
    },
    "Should create index using GEM (uncompressed references)": {
        "content": [
            "genome_index.gem:md5,ebd155f05251ace3a35ce3edd8816211",
            "transcript_index.junctions:md5,a0433e39f0c35e798609ff15332e12d4",
            "transcript_index.junctions.fa:md5,be0a28c75266e6969c11e308341810ae",
            "transcript_index.junctions.gem:md5,b53971e7b5358fd8cae8bb7f2a4a729d",
            "transcript_index.junctions.keys:md5,34985e55c280acd027d2dd32906ad59c"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.0"
        },
        "timestamp": "2024-05-02T09:02:27.226869622"
    }
}